src/kernel/disconnected_transactions.h

```diff
@@ -38,11 +41,28 @@ class DisconnectedBlockTransactions {
     /** Cached dynamic memory usage for the CTransactions (memory for the shared pointers is
      * included in the container calculations). */
     uint64_t cachedInnerUsage = 0;
+    const size_t m_max_mem_usage;
     std::list<CTransactionRef> queuedTx;
     using TxList = decltype(queuedTx);
     std::unordered_map<uint256, TxList::iterator, SaltedTxidHasher> iters_by_txid;
 
+    /** Trim the earliest-added entries until we are within memory bounds. */
+    std::vector<CTransactionRef> LimitMemoryUsage()
+    {
+        std::vector<CTransactionRef> evicted;
+
+        while (!queuedTx.empty() && DynamicMemoryUsage() > m_max_mem_usage) {
+            evicted.emplace_back(queuedTx.front());
+            cachedInnerUsage -= RecursiveDynamicUsage(*queuedTx.front());
+            iters_by_txid.erase(queuedTx.front()->GetHash());
+            queuedTx.pop_front();
+        }
+        return evicted;
+    }
+
 public:
+    DisconnectedBlockTransactions(size_t max_mem_usage) : m_max_mem_usage{max_mem_usage} {}
```

Comment:
nit: We are always passing the same `MAX_DISCONNECTED_TX_POOL_SIZE * 1000` as the constructor argument.
We can just declare a default value for it
```suggestion
    DisconnectedBlockTransactions(const size_t max_mem_usage = MAX_DISCONNECTED_TX_POOL_SIZE * 1000) : m_max_mem_usage{max_mem_usage} {}
```

Reply:
I'm not a fan of default arguments as it makes it easier to misuse functions. I think it's useful to keep this configurable and would prefer that a caller be aware of / explicit about bounding the memusage.